_source/_posts/2022-04-21-jhipster-k8s-azure.md

```diff
@@ -0,0 +1,667 @@
+
+If you get an error that says`(BadRequest) DNS record for cosmosdb under zone Document is already taken.` you need to change the `--name` parameter to something else.
+
+I'm using the Australia East location because that was the location that had free tier AKS nodes available when I wrote this tutorial. You can use any resource group you want as long as it allows you to create the AKS cluster later in the tutorial. Even if you can't use the free tier or the free credits, if you stop and start the AKS cluster between working on the tutorial, the cost should be very small (mine was less than a few dollars). The application should still work if the Cosmos DB database is in a different resource group and region since the database URI is configured to be publicly accessible.
+
+List the connection string for the Cosmos DB API for MongoDB endpoint using the following command.
+
+```bash
+az cosmosdb keys list --type connection-strings --name jhipster-cosmosdb --resource-group australia-east
+```
+
+This will list four connection strings. You need to save (copy and paste somewhere) the first, the primary connection string. (Ellipses have been used for brevity below)
+
+```bash
+  "connectionStrings": [
+    {
+      "connectionString": "mongodb://jhipster-cosmosdb:XBq5KZ81V8hM63KjCOezi1arq...,
+      "description": "Primary MongoDB Connection String"
+    },
+    ...
+  ]
+}
+```
+
+Create a `.env` file in the `docker-compose` subdirectory. Add the following variables to it, substituting your connection string for the placeholder. Make sure the connection string is enclosed in quotes. This value is referenced by the `docker-compose.yml` file and passed to the `store` service, pointing it to the Cosmos DB MongoDB database.
+
+The `ENCRYPT_KEY` will be used as the key for encrypting sensitive values stored in the Spring Cloud Config and used by the JHipster registry. You can put whatever value you want in there. A UUID works well, but any string value will work. The longer, the better.
+
+`docker-compose/.env`
+
+```.env
+SPRING_DATA_MONGO_URI="<your-connection-string>"
+ENCRYPT_KEY=<your-encryption-key
+```
+
+#### Configure Okta OAuth
+
+Use the Okta CLI to create an OIDC (OpenID Connect) application. This is what you need on the Okta side to use Okta as an authentication provider.
+
+If you already have an Okta account, use `okta login` to log into that account with the CLI. Otherwise, use `okta register` to sign up for a free account. 
+
+Create the OIDC app using the following command using a Bash shell opened to the project root.
+
+```bash
+okta apps create jhipster
+```
+
+You can accept the default values by pressing **enter**. If you want to give the app a different, more descriptive, name, that is fine. Just take note of the name because you will need to find the app in the Okta developer dashboard and update the redirect URIs a little later.
```

Comment:
Please try to use the [CLI include](https://github.com/oktadev/okta-blog#use-the-okta-cli-to-register-your-app) so we can change things in one place if we need to.

```
{% include setup/cli.md type="jhipster" %}
```